### PR TITLE
fix: [DEL-3370]: send proper error if token is already there

### PR DESCRIPTION
**Pull request by arpit3107**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30981)

### CODE REVIEW

File: 400-rest/src/main/java/io/harness/service/impl/DelegateTokenServiceImpl.java [hunk 1]
```diff
@@ -54,8 +56,11 @@ public DelegateTokenDetails createDelegateToken(String accountId, String name) {
                                       .value(Misc.generateSecretKey())
                                       .build();
 
-    persistence.save(delegateToken);
-
+    try {
+      persistence.save(delegateToken);
+    } catch (DuplicateKeyException e) {
+      throw new InvalidRequestException("Token with given name already exists for given account.");
```
Comment (abhinav-harness) [inline 400-rest/src/main/java/io/harness/service/impl/DelegateTokenServiceImpl.java:62]: token with name {} already exists.

**Review by abhinav-harness — COMMENTED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30981)* status: 0 of 1 approvals obtained.

**Review by abhinav-harness — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30981)* status: :shipit: complete! ✓

**Review by abhinav-harness — DISMISSED** [review]
> Reviewed 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30981)* status: :shipit: complete! ✓

**Review by Vbhatia20 — APPROVED** [review]
> Reviewed 1 of 4 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30981)* status: :shipit: complete! ✓

**Pull request merged**